src/testdir/test_findfile.vim

```diff
@@ -1,25 +1,167 @@
+  call assert_equal('Xdir2/Xdir3/bar',          findfile('bar', '**', 2))
+  call assert_equal('',                         findfile('bar', '**', 3))
+  call assert_equal(['bar', 'Xdir2/Xdir3/bar'], findfile('bar', '**', -1))
+
+  " Test upwards search.
+  cd Xdir2/Xdir3
+  call assert_equal('bar',                         findfile('bar', ';'))
+  call assert_equal(save_dir . '/Xdir1/Xdir2/foo', findfile('foo', ';'))
+  call assert_equal(save_dir . '/Xdir1/Xdir2/foo', findfile('foo', ';', 1))
+  call assert_equal(save_dir . '/Xdir1/foo',       findfile('foo', ';', 2))
+  call assert_equal(save_dir . '/Xdir1/foo',       findfile('foo', ';', 2))
+  call assert_equal(save_dir . '/Xdir1/Xdir2/foo', findfile('foo', 'Xdir2;', 1))
+  call assert_equal('',                            findfile('foo', 'Xdir2;', 2))
+
+  " List l should have at least 2 values (possibly more if foo file
+  " happens to be found upwards above Xdir1).
+  let l = findfile('foo', ';', -1)
+  call assert_equal(save_dir . '/Xdir1/Xdir2/foo', l[0])
+  call assert_equal(save_dir . '/Xdir1/foo',       l[1])
+
+  " Test upwards search with stop-directory.
+  cd Xdir2
+  let l = findfile('bar', ';' . save_dir . '/Xdir1/Xdir2/', -1)
+  call assert_equal([save_dir . '/Xdir1/Xdir2/Xdir3/bar'], l)
+
+  let l = findfile('bar', ';' . save_dir . '/Xdir1/', -1)
+  cal assert_equal([ save_dir . '/Xdir1/Xdir2/Xdir3/bar',
+  \                  save_dir . '/Xdir1/bar' ], l)
+
+  " Test combined downwards and upwards search from Xdir2/.
+  cd ../..
+  call assert_equal('Xdir3/bar',             findfile('bar', '**;', 1))
+  call assert_equal(save_dir . '/Xdir1/bar', findfile('bar', '**;', 2))
+
+  bwipe!
+  exe 'cd  ' . save_dir
+  call CleanFiles()
+  let &path = save_path
+  let &shellslash = save_shellslash
+endfunc
+
+" Test finddir({name} [, {path} [, {count}]])
+func Test_finddir()
+  let save_path = &path
+  let save_shellslash = &shellslash
+  let save_dir = getcwd()
+  set path=,,
+  call CreateFiles()
+  cd Xdir1
+
+  call assert_equal('Xdir2', finddir('Xdir2'))
+  call assert_equal('',      finddir('Xdir3'))
+
+  " Files should not be found (findfile() finds them).
+  call assert_equal('', finddir('foo'))
+
+  call assert_equal('Xdir2',       finddir('Xdir2', '**'))
+  call assert_equal('Xdir2/Xdir3', finddir('Xdir3', '**'))
+
+  call assert_equal('Xdir2',               finddir('Xdir2', '**', 1))
+  call assert_equal('Xdir2/Xdir3/Xdir2',   finddir('Xdir2', '**', 2))
+  call assert_equal(['Xdir2',
+        \            'Xdir2/Xdir3/Xdir2'], finddir('Xdir2', '**', -1))
+
+  call assert_equal('Xdir2',       finddir('Xdir2', '**1'))
+  call assert_equal('Xdir2',       finddir('Xdir2', '**0'))
+  call assert_equal('Xdir2/Xdir3', finddir('Xdir3', '**1'))
+  call assert_equal('',            finddir('Xdir3', '**0'))
+
+  " Test upwards dir search.
+  cd Xdir2/Xdir3
+  call assert_equal(save_dir . '/Xdir1', finddir('Xdir1', ';'))
```

Comment:
This fails on Windows in CI:
```
Test_finddir line 29: Expected 'C:/projects/vim/src/testdir/Xdir1' but got '/projects/vim/src/testdir/Xdir1'
```
I'm not well familiar with Windows, but it does not seem to me
that these 2 full file names are equivalent. Isn't this a bug
with finddir()?  `finddir('Xdir1', ';')` is supposed to return a full
path when searching upwards, so I would expect it to have C:
as in the expectation of the test. 

I could use `assert_match` instead of `assert_equal` to ignore this
difference but I first want to make sure it's not a bug.